src/activity_item_handling.cpp

```diff
@@ -932,433 +939,1123 @@ static construction check_build_pre( const construction &con )
-                // and unload them
-                p.assign_activity( act_multiple_construction );
-                p.set_moves( 0 );
-                g->reload_npcs();
-                return;
+static bool are_requirements_nearby( const std::vector<tripoint> &loot_spots,
+                                     const requirement_id &needed_things, const player &p, const activity_id activity_to_restore,
+                                     bool in_loot_zones )
+{
+    zone_manager &mgr = zone_manager::get_manager();
+    inventory temp_inv;
+    units::volume volume_allowed = p.volume_capacity() - p.volume_carried();
+    units::mass weight_allowed = p.weight_capacity() - p.weight_carried();
+    for( const tripoint &elem : loot_spots ) {
+        // if we are searching for things to fetch, we can skip certain thngs.
+        // if, however they are already near the work spot, then the crafting / inventory fucntions will have their own method to use or discount them.
+        if( in_loot_zones ) {
+            // skip tiles in IGNORE zone and tiles on fire
+            // (to prevent taking out wood off the lit brazier)
+            // and inaccessible furniture, like filled charcoal kiln
+            if( mgr.has( zone_type_id( "LOOT_IGNORE" ), g->m.getabs( elem ) ) ||
+                g->m.dangerous_field_at( elem ) ||
+                !g->m.can_put_items_ter_furn( elem ) ) {
+                continue;
             }
-            const std::vector<tripoint> route = route_adjacent( p, src_loc );
-            if( route.empty() ) {
-                // can't get there, can't do anything, skip it
+        }
+        for( const auto elem2 : g->m.i_at( elem ) ) {
+            if( in_loot_zones && elem2.made_of_from_type( LIQUID ) ) {
                 continue;
             }
-            p.set_destination( route, player_activity( act_multiple_construction ) );
-            return;
+            if( p.backlog.front().id() == activity_id( "ACT_MULTIPLE_FARM" ) ||
+                activity_to_restore == activity_id( "ACT_MULTIPLE_FARM" ) ) {
+                // this fetch task will need to pick up an item. so check for its weight/volume before setting off.
+                if( in_loot_zones && ( elem2.type->volume > volume_allowed ||
+                                       elem2.type->weight > weight_allowed ) ) {
+                    continue;
+                }
+            }
+            temp_inv += elem2;
         }
-        // dont go there if it's dangerous.
-        bool dangerous_field = false;
-        for( const std::pair<const field_type_id, field_entry> &e : g->m.field_at( src_loc ) ) {
-            if( p.is_dangerous_field( e.second ) ) {
-                dangerous_field = true;
-                break;
+        if( !in_loot_zones ) {
+            if( const cata::optional<vpart_reference> vp = g->m.veh_at( elem ).part_with_feature( "CARGO",
+                    false ) ) {
+                vehicle *src_veh = &vp->vehicle();
```

Comment:
```suggestion
                vehicle &src_veh = vp->vehicle();
```
Again: what's with the pointers?